packages/twenty-front/src/modules/favorites/hooks/useDeleteFavoriteFolder.ts

```diff
@@ -0,0 +1,26 @@
+import { CoreObjectNameSingular } from '@/object-metadata/types/CoreObjectNameSingular';
+import { useDeleteOneRecord } from '@/object-record/hooks/useDeleteOneRecord';
+import { usePrefetchedFavoritesData } from './usePrefetchedFavoritesData';
+
+export const useDeleteFavoriteFolder = () => {
+  const { deleteOneRecord } = useDeleteOneRecord({
+    objectNameSingular: CoreObjectNameSingular.FavoriteFolder,
+  });
+  const { upsertFavorites, favorites, workspaceFavorites } =
+    usePrefetchedFavoritesData();
```

Comment:
this is a bit a waste! useDeleteFavoriteFolder should expose a synchronous function. Here we are listening to favorites (this will trigger a re-render of the hook and of any component using the hook) but we only need to know the favorites during the sync action deleteFavoriteFolder. We should use a recoilCallback to read from the state